atomic_reactor/plugins/exit_pulp_publish.py

```diff
@@ -0,0 +1,122 @@
+
+However, the recommended scheme (since Origin 1.0.6) is to store a
+key and certificate in a secret which the builder service account is
+allowed to mount:
+
+$ oc secrets new pulp ./pulp.key ./pulp.cer
+secrets/pulp
+$ oc secrets add serviceaccount/builder secret/pulp --for=mount
+
+In the BuildConfig for atomic-reactor, specify the secret in the
+strategy's 'secrets' array, specifying a mount path:
+
+"secrets": [{
+  "secretSource": {
+    "name": "pulp"
+  },
+  "mountPath": "/var/run/secrets/pulp"
+}]
+
+In the configuration for this plugin, specify the same path for
+pulp_secret_path:
+
+"pulp_publish": {
+  "pulp_registry_name": ...,
+  ...
+  "pulp_secret_path": "/var/run/secrets/pulp"
+}
+"""
+
+from __future__ import print_function, unicode_literals
+
+from atomic_reactor.constants import PLUGIN_PULP_PUBLISH_KEY
+from atomic_reactor.plugins.build_orchestrate_build import get_worker_build_info
+from atomic_reactor.plugin import ExitPlugin
+from atomic_reactor.util import ImageName
+from atomic_reactor.pulp_util import PulpHandler
+
+
+class PulpPublishPlugin(ExitPlugin):
+    key = PLUGIN_PULP_PUBLISH_KEY
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, pulp_registry_name,
+                 pulp_secret_path=None, username=None, password=None,
+                 dockpulp_loglevel=None):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param pulp_registry_name: str, name of pulp registry to use, specified in /etc/
+                                   dockpulp.conf
+        :param pulp_secret_path: path to pulp.cer and pulp.key; $SOURCE_SECRET_PATH otherwise
+        :param username: pulp username, used in preference to certificate and key
+        :param password: pulp password, used in preference to certificate and key
+        """
+        # call parent constructor
+        super(PulpPublishPlugin, self).__init__(tasker, workflow)
+        self.pulp_registry_name = pulp_registry_name
+        self.pulp_secret_path = pulp_secret_path
+        self.username = username
+        self.password = password
+        self.dockpulp_loglevel = dockpulp_loglevel
+        self.pulp_handler = PulpHandler(self.workflow, self.pulp_registry_name, self.log,
+                                        pulp_secret_path=self.pulp_secret_path,
+                                        username=self.username, password=self.password,
+                                        dockpulp_loglevel=self.dockpulp_loglevel)
+
+    def publish_to_crane(self, repo_prefix="redhat-"):
+        image_names = self.workflow.tag_conf.images[:]
+        # Find out how to publish this image.
+        self.log.info("image names: %s", [str(image_name) for image_name in image_names])
+
+        pulp_repos = self.pulp_handler.create_dockpulp_and_repos(image_names, repo_prefix)
+
+        self.pulp_handler.publish(pulp_repos.keys())
```

Comment:
I think this beginning section should read:

```
pulp_repos = set([image.pulp_repo for image in self.workflow.tag_conf.images])
self.pulp_handler.publish(pulp_repos)
```

`.pulp_repo` is an attribute of a tag_conf 'image', i.e. if the tag is named foo/bar:baz then the "pulp_repo" for it is named redhat-foo-bar. We're just trying to find out which repositories need to get published, given the tag_conf.

Note: tag_conf will be set by the group_manifests plugin.